WordPress/Classes/Utility/NotificationRemindersHelper.swift

```diff
@@ -0,0 +1,276 @@
+            content.body = subject
+        }
+        content.categoryIdentifier = RequestKeys.category
+        content.sound = UNNotificationSound.default
+        content.userInfo = [
+            RequestKeys.type: RequestKeys.typeValue,
+            RequestKeys.notificationId: note.notificationId
+        ]
+
+        let trigger = UNCalendarNotificationTrigger(dateMatching: components, repeats: false)
+        let identifier = UUID().uuidString
+        let notificationRequest = UNNotificationRequest(identifier: identifier,
+                                                        content: content,
+                                                        trigger: trigger)
+
+        UNUserNotificationCenter.current().add(notificationRequest, withCompletionHandler: { error in
+            if let error = error {
+                DDLogError("Unable to add reminder notification request (\(error), \(error.localizedDescription))")
+                completion?(false)
+            } else {
+                self.trackReminder(withId: identifier)
+                completion?(true)
+            }
+        })
+    }
+
+    // MARK: - Get Reminder Details
+
+    /// - returns: The Notification title for the specified reminder request.
+    ///
+    func reminderTitle(for request: UNNotificationRequest) -> String {
+        guard let noteId = reminderNotificationId(from: request.content.userInfo),
+            let note = loadNotification(with: noteId),
+            let subject = note.renderSubject()?.string else {
+                return ""
+        }
+
+        return subject
+    }
+
+    /// - returns: A Notification identifier contained within the provided userInfo,
+    ///   if one is present.
+    ///
+    func reminderNotificationId(from userInfo: [AnyHashable: Any]) -> String? {
+        return userInfo[RequestKeys.notificationId] as? String
+    }
+
+    /// - returns: The trigger date for the specified reminder request.
+    ///
+    func reminderTriggerDate(for request: UNNotificationRequest) -> Date? {
+        let trigger = request.trigger as? UNCalendarNotificationTrigger
+        return trigger?.nextTriggerDate()
+    }
+
+    /// Fetches the Notification with the specified ID from Core Data.
+    ///
+    private func loadNotification(with noteId: String) -> Notification? {
+        let predicate = NSPredicate(format: "(notificationId == %@)", noteId)
+
+        return ContextManager.shared.mainContext.firstObject(ofType: Notification.self,
+                                                             matching: predicate)
+    }
+
+    // MARK: - Managing Reminders
+
+    /// Logs the specified reminder ID, used when fetching and filtering
+    /// pending reminders. A reminder ID must be tracked using this
+    /// method for it to be returned by `getPendingReminders`.
+    ///
+    /// - parameter id: The identifier of a UNNotificationRequest
+    ///
+    func trackReminder(withId id: String) {
```

Comment:
Is this function expected to be used outside of this helper class and getPendingReminders or can it be private?